### ISSUE TITLE
Fix request: false positive for OptimizeMapOrSetForEnum

### ISSUE THREAD

**Issue by rselie** [jborgers/PMD-jPinpoint-rules#194]
The OptimizeMapOrSetForEnum rule triggers on some code which I don't expect:

First on .collect(Maps.toImmutableEnumMap(
						Package::getPackageCode, packageConfig -> getPackageConfigSpecials(packageConfig)));
						
Where Maps is of com.google.common.collect.Maps.

Secondly when an interface is used to get a map it is also triggered. 
Map<PackageCode, List<X>> packageCodeMap = interface.getPackageCodeMap();

**Comment by jborgers** [comment]
Hi @rselie, could you please add some example code we can use as test, code which compiles?

**Comment by jborgers** [comment]
example code:
```
class Foo {
    private final BarGenerator barGenerator;

    public Foo(BarGenerator barCodeListMap) {
        this.barGenerator = barCodeListMap;
    }

    public Map<BarCode, List<String>> bar() {
        Map<BarCode, List<String>> barCodeListMap = barGenerator.generateBarCodeListMap(); // false positive
        // do something
        return barCodeListMap;
    }

    public void foo(List<BarConfig> barConfigs){
        Map<BarCode, String> barCodeDescriptionMap = barConfigs.stream() // false positive
                .collect(Maps.toImmutableEnumMap(BarConfig::getBarCode, BarConfig::getDescription));
    }
}

@Data
class BarConfig {
    private BarCode barCode;
    private String description;
}

interface BarGenerator {
    Map<BarCode, List<String>> generateBarCodeListMap();
}

enum BarCode {
    CHOCOLATE,
    NUTS
}

```

**Comment by jborgers** [comment]
When dependency, the interface is in a different class, there is still a false-positive. Then the type is known from the classpath, it then is actually exactly a Set or Map.
